src/Game/GamePage.js

```diff
@@ -37,6 +54,14 @@ const GamePage = (props) => {
     return totalPlayerData.find((playerData) => playerData.name === playerName);
   }, [totalPlayerData]);
 
+  const turnPlayer = useMemo(() => {
+    return totalPlayerData.find(
+      (playerData, index) => index === currentTurnPlayer
+    );
+  }, [currentTurnPlayer, totalPlayerData]);
```

Comment:
Naming is confusing. `turnPlayer` probably makes more sense to be `currentTurnPlayer` and `currentTurnPlayer` makes more sense to be `currentTurnPlayerIndex`